### PR TITLE
Fixes #154: Interpretation of Ruby source files no longer depends on …

### LINKED ISSUE #154: encoding detection breaks in locale "C" with UTF-8 source files
Since Ruby 2.0, source files are UTF-8 by default.

ruby-lint breaks with that in locale "C":

```
...lib/ruby/gems/2.2.0/gems/parser-2.2.2.6/lib/parser/source/buffer.rb:157:in `source=': invalid byte sequence in US-ASCII (EncodingError)
	from ...lib/ruby/gems/2.2.0/gems/ruby-lint-2.0.4/lib/ruby-lint/parser.rb:40:in `parse'
	from ...lib/ruby/gems/2.2.0/gems/ruby-lint-2.0.4/lib/ruby-lint/runner.rb:78:in `parse_file'
```

The reason is that File.read(file).encoding yields US-ASCII in locale "C".  reencode_string takes that for face value instead of properly substituting UTF-8.  (The logic in there is somewhat surprising -- clearly any source file without an encoding comment is UTF-8, independent of what File.read says.)

### PR DESCRIPTION
**Pull request by cabo**
…what happens to be in the locale.

(whitequark is correct in: https://github.com/whitequark/parser/issues/208)

### CODE REVIEW

**Comment by cabo** [comment]
(Actually, builds for all Ruby versions succeeded; there just seems to be a rvm problem on Macs that breaks before even getting to ruby-lint.)

File: lib/ruby-lint/runner.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def analyze_file(file, parser, report)
     # @return [Array]
     #
     def parse_file(parser, file)
-      return parser.parse(File.read(file), file)
+      return parser.parse(File.read(file, encoding: Encoding::UTF_8), file)
```
Comment (YorickPeterse) [inline lib/ruby-lint/runner.rb]: Please use the Hash rocket syntax, this keeps it consistent with other code.

**Comment by cabo** [comment]
BTW, if you want to faithfully mirror the way `::Parser::CurrentRuby` uses the currently running version of Ruby to select the language dialect, you need

```ruby
File.read(..., :encoding => RUBY_VERSION[0] == "1" ? Encoding::US_ASCII : Encoding::UTF_8)
```

but then I don't know how much of a feature that selection actually is for ruby-lint.

**Comment by YorickPeterse** [comment]
I'm not really sure why it would have to use US_ASCII for Ruby 1.9, UTF8 should work just fine. Either way, we'll go with this and see what happens. Thanks!

**Comment by YorickPeterse** [comment]
This fix is available in ruby-lint 2.0.5, which has just been pushed to RubyGems.

**Pull request merged**